recipe/upload_or_check_non_existence.py

```diff
@@ -99,28 +89,31 @@ def main():
     parser.add_argument('recipe_dir', help='the conda recipe directory')
     parser.add_argument('owner', help='the binstar owner/user')
     parser.add_argument('--channel', help='the binstar channel', default='main')
+    parser.add_argument("-m", "--variant-config-files", action="append",
+                        help="path to conda_build_config.yaml defining your base matrix")
     args = parser.parse_args()
     recipe_dir, owner, channel = args.recipe_dir, args.owner, args.channel
 
     cli = get_binstar(argparse.Namespace(token=token, site=None))
-    meta_main = MetaData(recipe_dir)
-    for _, meta in meta_main.get_output_metadata_set(files=None):
+    metas = conda_build.api.render(recipe_dir, variant_config_files=args.variant_config_files)
```

Comment:
I'm guessing `variant_config_files` will now be an empty list for existing feedstocks. How does this affect `metas`? Is `variant_config_files` simply ignored then?

Reply:
Yes, it's ignored.  Metas will lose their pins, if any, and they'll use default values for a few things.  Any explicit template values that are missing will cause errors.  Defaults are defined at https://github.com/conda/conda-build/blob/master/conda_build/variants.py#L22

Reply:
Ok. Thanks for the link.

Getting into the weeds a bit, but `cpu_optimization_target` is used to set the lower bound for the CPU target or will it actually result in dropping unsupported intrinsics for that target (e.g. vector intrinsics)?

Reply:
Actually I'm not seeing `cpu_optimization_target` anywhere else in `conda-build` is this used by something else? `{{ compiler() }}` presumably?

Reply:
Ok, will ignore it then. Thanks.